aws/resource_aws_lb_listener_rule.go

```diff
@@ -920,3 +1234,115 @@ func highestListenerRulePriority(conn *elbv2.ELBV2, arn string) (priority int64,
 
 	return
 }
+
+func lbListenerRuleConditions(conditions []interface{}) ([]*elbv2.RuleCondition, error) {
+	elbConditions := make([]*elbv2.RuleCondition, len(conditions))
+	for i, condition := range conditions {
+		conditionMap := condition.(map[string]interface{})
+		conditionValues := conditionMap["values"].([]interface{})
+
+		elbConditions[i] = &elbv2.RuleCondition{
+			Field: aws.String(conditionMap["field"].(string)),
+		}
+		switch conditionMap["field"].(string) {
+		case "host-header":
+			hostHeader := conditionMap["host_header"].([]interface{})
```

Comment:
Don't we know the type of it already, i.e. it being a string?
```suggestion
			hostHeader := conditionMap["host_header"].([]string{})
```

Thoughts?

Reply:
As above, a data structure generated by Terraform is all `interface{}` 😞